projects/postgresql/fuzzer/protocol_fuzzer.c

```diff
@@ -0,0 +1,116 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include "postgres.h"
+
+#include "access/xlog.h"
+#include "access/xact.h"
+#include "common/ip.h"
+#include "common/username.h"
+#include "executor/spi.h"
+#include "jit/jit.h"
+#include "libpq/auth.h"
+#include "libpq/libpq.h"
+#include "libpq/pqsignal.h"
+#include "miscadmin.h"
+#include "optimizer/optimizer.h"
+#include "parser/analyze.h"
+#include "parser/parser.h"
+#include "storage/proc.h"
+#include "tcop/tcopprot.h"
+#include "utils/datetime.h"
+#include "utils/memutils.h"
+#include "utils/memdebug.h"
+#include "utils/pidfile.h"
+#include "utils/portal.h"
+#include "utils/snapmgr.h"
+#include "utils/ps_status.h"
+#include "utils/timeout.h"
+
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <libgen.h>
+
+const char *progname = "progname";
+static sigjmp_buf postgre_exit;
+static bool postgre_started;
+static char *buffer;
+static size_t buffersize;
+static char *bufferpointer;
+static char *av[6];
+
+int LLVMFuzzerInitialize(int *argc, char ***argv) {
+	char *exe_path = (*argv)[0];
+	//dirname() can modify its argument
+	char *exe_path_copy = strdup(exe_path);
+	char path_to_binary[100];
+	char *dir = dirname(exe_path_copy);
```

Comment:
i just had an idea: just chdir to the result of dirname, we can have everything relative then...